### PR TITLE
[DNM until 2.7.2 is released] [RFE] Agent deployment customization support: default structs and settings

### PR DESCRIPTION
**Pull request by a-blender**
## Issue: 

https://github.com/rancher/rancher/issues/41035
 
## Problem

Agent deployment customization support for fleet and cluster agent for customers to be able to configure agent pod scheduling and resources.
 
## Solution

- [x] Data structures and new settings for defaults. Includes:
	- `AgentDeploymentCustomization` struct and adding it to both provisioning and management Cluster objects
	- `cluster-agent-default-affinity` setting with the default value matching the current affinity used for `cluster-agent` deployment
	- `fleet-agent-default-affinity` setting with the default value matching the current affinity used for `fleet-agent` deployment
 
### Manual Testing

Verified `go generate` runs without error

### CODE REVIEW

File: pkg/apis/management.cattle.io/v3/cluster_types.go
Comment (snasovich) [inline pkg/apis/management.cattle.io/v3/cluster_types.go:128]: We want to define separate fields for cattle-cluster-agent and fleet-agent, as below:
```
ClusterAgentDeploymentCustomization *AgentDeploymentCustomization `json:"clusterAgentDeploymentCustomization,omitempty"`
FleetAgentDeploymentCustomization   *AgentDeploymentCustomization `json:"fleetAgentDeploymentCustomization,omitempty"`
```

**Review by snasovich — COMMENTED** [review]
Leaving some preliminary feedback. Good job, seems to be very close to what we need to unblock UI!

**Review by snasovich — DISMISSED** [review]
LGTM! Waiting for 2.7.2 release before we merge this one.

**Comment by snasovich** [comment]
Actually, circling back on this it looks like the new fields are not on the `cluster.provisioning.cattle.io` object and should be there too.

File: pkg/settings/setting.go
Comment (Oats87) [inline pkg/settings/setting.go:242]: Are we going to start using these default affinity values when we start rendering the `cattle-cluster-agent` manifests?

File: pkg/settings/setting.go
Comment (Oats87) [inline pkg/settings/setting.go:338]: I worry that these are going to fall out of date -- what kind of process are we going to put in place to keep the defaults we set here updated with what Fleet defines as their default values?

**Review by Oats87 — CHANGES_REQUESTED** [review]
@snasovich @a-blender I'm not entirely sure why we are opening a PR to merge API/struct changes (and settings changes) without the corresponding plumbing for these functions.

It's difficult to review this PR in its current state because there could be gotchas that aren't found until we actually plumb the new fields in.

File: pkg/settings/setting.go
Comment (Oats87) [reply pkg/settings/setting.go:242]: Can we move this stuff out into its own file in the same package? In its current state, it makes an already difficult-to-grok file even bigger.

Comment (a-blender) [reply pkg/settings/setting.go:242]: I believe so yes. Those default values will be set in the cluster-agent manifest if no override is provided. We could, but as I understand it when we do the backend plumbing I'll move this default affinity object code out anyway and the settings will pull the default affinity via a middle function.

File: pkg/settings/setting.go
Comment (a-blender) [reply pkg/settings/setting.go:338]: May more be a question for the Fleet team

Comment (snasovich) [reply pkg/settings/setting.go:338]: That is part of the current design and I pinged Fleet devs to check on the process to keep these updated.
Luckily default affinity didn't change in the last 2 years or so (now that I said it it's probably going to change next month).

File: pkg/settings/setting.go
Comment (snasovich) [reply pkg/settings/setting.go:242]: @a-blender , this matches what I brought up in our offline sync last week.
The logic that actually renders this default affinity into YAML manifest will likely need object/YAML representation of it while setting is in JSON. So it makes sense to have a func that returns actual `Affinity` object so it can be used in both places.

But that's not important to address as part of this "data model" PR.

File: pkg/apis/provisioning.cattle.io/v1/cluster.go
Comment (snasovich) [reply pkg/apis/provisioning.cattle.io/v1/cluster.go:36]: Just curious, what's the logic by which these fields are grouped in the blocks? Would be great to add comments to explain.

**Comment by snasovich** [comment]
@a-blender , I guess we want to close this one in favor of https://github.com/rancher/rancher/pull/41203. If correct, please close it.

**Pull request closed without merging**
